engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -475,35 +687,46 @@ namespace dmGameSystem
                     if (info->m_Attribute->m_CoordinateSpace == dmGraphics::COORDINATE_SPACE_WORLD)
                     {
                         Vector4 wp = w * p;
-                        memcpy(vertices_write_ptr, &wp, info->m_ValueByteSize);
+                        memcpy(write_ptr, &wp, info->m_ValueByteSize);
                     }
                     else if (info->m_Attribute->m_CoordinateSpace == dmGraphics::COORDINATE_SPACE_LOCAL)
                     {
-                        memcpy(vertices_write_ptr, &p_local, info->m_ValueByteSize);
+                        memcpy(write_ptr, &p_local, info->m_ValueByteSize);
                     }
                     else assert(0);
                 } break;
                 case dmGraphics::VertexAttribute::SEMANTIC_TYPE_TEXCOORD:
                 {
-                    memcpy(vertices_write_ptr, uv, info->m_ValueByteSize);
+                    uint32_t unit = num_texcoords++;
+                    if (unit >= num_textures)
+                        unit = 0;
+                    memcpy(write_ptr, uvs[unit].Begin()+vertex_index*2, info->m_ValueByteSize);
                 } break;
                 case dmGraphics::VertexAttribute::SEMANTIC_TYPE_PAGE_INDEX:
                 {
-                    memcpy(vertices_write_ptr, &page_index, info->m_ValueByteSize);
+                    uint32_t unit = num_page_indices++;
+                    memcpy(write_ptr, &page_indices[unit], info->m_ValueByteSize);
                 } break;
                 default:
                 {
-                    memcpy(vertices_write_ptr, info->m_ValuePtr, info->m_ValueByteSize);
+                    memcpy(write_ptr, info->m_ValuePtr, info->m_ValueByteSize);
                 } break;
             }
+        }
+    }
 
-            vertices_write_ptr += info->m_ValueByteSize;
+    static void EnsureSize(dmArray<float>& array, uint32_t size)
+    {
+        if (array.Capacity() < size) {
+            array.OffsetCapacity(size - array.Capacity());
         }
+        array.SetSize(size);
     }
```

Comment:
Maybe this should be a first class function in dmarray, we use this pattern alot afaik